### PR TITLE
test: `VerifyClientMessage` using mockVM

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #4823 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/types/misbehaviour_handle_test.go [hunk 1]
```diff
@@ -1,499 +1,92 @@
...
-				data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-				// VerifyClientMessage must be run first
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
-				clientState.UpdateState(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
+				suite.mockVM.RegisterQueryCallback(types.VerifyClientMessageMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/types/misbehaviour_handle_test.go:30]: same recommendation as other PR, we should unmarshal `queryMsg []byte` into the expected type and make some assertions in here

File: modules/light-clients/08-wasm/types/misbehaviour_handle_test.go [hunk 2]
```diff
@@ -9,505 +9,100 @@ import (
...
 	}
 
 	for _, tc := range testCases {
 		suite.Run(tc.name, func() {
-			suite.SetupWasmGrandpaWithChannel()
-			clientState, ok = suite.chainA.App.GetIBCKeeper().ClientKeeper.GetClientState(suite.ctx, defaultWasmClientID)
-			suite.Require().True(ok)
+			// reset suite to create fresh application state
+			suite.SetupWasmWithMockVM()
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
```
Comment (DimitrisJim) [inline modules/light-clients/08-wasm/types/misbehaviour_handle_test.go:81]: inb4 damian spacing comment

```suggestion
			suite.SetupWasmWithMockVM()

			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
```

File: modules/light-clients/08-wasm/types/misbehaviour_handle_test.go [hunk 3]
```diff
@@ -9,505 +9,100 @@ import (
...
+					suite.Require().NotNil(msg.VerifyClientMessage.ClientMessage)
+					suite.Require().Nil(msg.Status)
+					suite.Require().Nil(msg.CheckForMisbehaviour)
+					suite.Require().Nil(msg.TimestampAtHeight)
+					suite.Require().Nil(msg.ExportMetadata)
+
+					resp, err := json.Marshal(types.EmptyResult{})
+					if err != nil {
+						return nil, 0, err
+					}
```
Comment (DimitrisJim) [inline modules/light-clients/08-wasm/types/misbehaviour_handle_test.go:45]: I'd go with:

```suggestion
					resp, err := json.Marshal(types.EmptyResult{})
					suite.Require().NoError(err)
```

**Review by DimitrisJim — APPROVED** [review]
l g t m

**Review by damiannolan — APPROVED** [review]
Noice! LGTM 🚀

**Pull request merged**